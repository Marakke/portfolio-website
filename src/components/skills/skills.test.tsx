import React from "react";
import { render } from "@testing-library/react";
import * as Gatsby from "gatsby";
import Skills from "./index";
import '../../../__mocks__/intersectionObserverMock';

const useStaticQuery = jest.spyOn(Gatsby, `useStaticQuery`);

const mockUseStaticQuery = {
  skillset: {
    title: "Test Title",
    skills: [
      {
        name: "Test Name",
        image: {
          fluid: () => {
            return {
              aspectRatio: 1,
              src: "mock/image/path.jpg",
              srcSet: "mock/srcSet",
              sizes: "(max-width: 800px) 100vw, 800px",
            };
          },
          alt: "Test Alt Text",
        },
        level: "Level:",
        xp: "XP:"
      }
    ],
    helpText: "Test Help Text",
  },
};

describe("Skills", () => {
  beforeEach(() => {
    useStaticQuery.mockImplementation(() => mockUseStaticQuery);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("renders the Skills component correctly", () => {
    const { container, getByTestId, getByAltText, getByRole } = render(<Skills />);
    expect(container).toMatchSnapshot();

    const skillsContainer = getByTestId("skills-container");
    expect(skillsContainer).toBeInTheDocument();

    const titleElement = getByRole("heading", { name: "Test Title" });
    expect(titleElement).toBeInTheDocument();

    const imageElement = getByAltText("Test Alt Text");
    expect(imageElement).toBeInTheDocument();

    const paragraphElements = container.querySelectorAll("p");
    expect(paragraphElements.length).toEqual(7);
    expect(paragraphElements[0].textContent).toBe("Level:");
    expect(paragraphElements[1].textContent).toBe("XP:");
  });
});
