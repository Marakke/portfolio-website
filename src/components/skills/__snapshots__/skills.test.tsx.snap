// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills renders the Skills component correctly 1`] = `
<div>
  <div
    class="skills-container"
    data-testid="skills-container"
    id="skills"
  >
    <div
      class="skills-content"
    >
      <h2>
        Test Title
      </h2>
      <div
        class="card-carousel-wrapper"
      >
        <div
          class="card-carousel"
        >
          <div
            class="card-carousel-card"
          >
            <div
              class="skill-card"
              tabindex="0"
            >
              <div
                class=" gatsby-image-wrapper"
                style="position: relative; overflow: hidden;"
              >
                <div
                  aria-hidden="true"
                  style="width: 100%;"
                />
                <picture>
                  <img
                    alt="Test Alt Text"
                    loading="lazy"
                    style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
                  />
                </picture>
                <noscript>
                  &lt;picture&gt;&lt;source srcset="undefined" /&gt;&lt;img loading="lazy" src="" alt="Test Alt Text" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
                </noscript>
              </div>
              <div
                class="skill-details"
              >
                <h3>
                  Test Name
                </h3>
                <div
                  class="detail"
                >
                  <p
                    class="level"
                  >
                    Level:
                  </p>
                   
                  Level:
                </div>
                <div
                  class="detail"
                >
                  <p
                    class="xp"
                  >
                    XP:
                  </p>
                   
                  XP:
                   years
                  <img
                    alt="Question Icon"
                    src="test-file-stub"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="card-carousel-button-right"
          data-testid="button-right"
        >
          <span>
            &gt;
          </span>
        </button>
      </div>
    </div>
    <div
      class="lines horizontal"
    >
      <p
        class="line horizontal"
      />
      <p
        class="line horizontal"
      />
      <p
        class="line horizontal"
      />
      <p
        class="line horizontal"
      />
      <p
        class="line horizontal"
      />
    </div>
  </div>
</div>
`;
